### ISSUE TITLE
Doesn't compile with musl

### ISSUE THREAD

**Issue by spease** [DoumanAsh/lazy-socket.rs#5]
Saw the deprecation warning, unfortunately this is still used by tokio_ping.

When I try to compile with a musl target, I get the following errors:
```
error[E0308]: mismatched types
   --> /home/spease/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy-socket-0.2.1/src/raw/unix.rs:458:37
    |
458 |             match ioctl(self.inner, request, value) {
    |                                     ^^^^^^^ expected i32, found u64

error[E0308]: mismatched types
   --> /home/spease/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy-socket-0.2.1/src/raw/unix.rs:467:20
    |
467 |         self.ioctl(FIONBIO, (!value) as c_ulong)
    |                    ^^^^^^^ expected u64, found i32

error: aborting due to 2 previous errors

```

**Comment by DoumanAsh** [comment]
I'll take a look at `musl` definition of `FIONBIO` and `ioctl`

**Comment by DoumanAsh** [comment]
Can you please check https://github.com/DoumanAsh/lazy-socket.rs/commit/43f67d226ffa36fc28da09135aa51c01a03724e3 ?

**Comment by spease** [comment]
That fixes it - thanks so much!

**Comment by DoumanAsh** [comment]
Ok, I'll release new version later on today

**Comment by DoumanAsh** [comment]
@spease Published as [0.2.2](https://crates.io/crates/lazy-socket/0.2.2)
There is also [0.3.0](https://crates.io/crates/lazy-socket/0.3.0) with new winapi 0.3

**Comment by DoumanAsh** [comment]
@spease P.s. latest `tokio_ping` doesn't depend on this library and instead uses socket2
